components/camel-huaweicloud-functiongraph/src/main/java/org/apache/camel/FunctionGraphProducer.java

```diff
@@ -0,0 +1,246 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.huaweicloud.sdk.core.auth.BasicCredentials;
+import com.huaweicloud.sdk.core.http.HttpConfig;
+import com.huaweicloud.sdk.functiongraph.v2.FunctionGraphClient;
+import com.huaweicloud.sdk.functiongraph.v2.model.InvokeFunctionRequest;
+import com.huaweicloud.sdk.functiongraph.v2.model.InvokeFunctionResponse;
+import com.huaweicloud.sdk.functiongraph.v2.region.FunctionGraphRegion;
+import org.apache.camel.constants.FunctionGraphConstants;
+import org.apache.camel.constants.FunctionGraphOperations;
+import org.apache.camel.constants.FunctionGraphProperties;
+import org.apache.camel.models.ClientConfigurations;
+import org.apache.camel.support.DefaultProducer;
+import org.apache.camel.util.ObjectHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class FunctionGraphProducer extends DefaultProducer {
+    private static final Logger LOG = LoggerFactory.getLogger(FunctionGraphProducer.class);
+    private FunctionGraphEndpoint endpoint;
+    private ClientConfigurations clientConfigurations;
+    private FunctionGraphClient functionGraphClient;
+    private BasicCredentials auth;
+    private HttpConfig httpConfig;
+
+    @Override
+    protected void doStart() throws Exception {
+        super.doStart();
+        this.clientConfigurations = new ClientConfigurations(this.endpoint);
+        initClient();
+    }
+
+    public FunctionGraphProducer(FunctionGraphEndpoint endpoint) {
+        super(endpoint);
+        this.endpoint = endpoint;
+    }
+
+    public void process(Exchange exchange) throws Exception {
+        updateClientConfigs(exchange);
+
+        switch (clientConfigurations.getOperation()) {
+            case FunctionGraphOperations.INVOKE_FUNCTION:
+                invokeFunction(exchange);
+                break;
+            default:
+                throw new UnsupportedOperationException(String.format("%s is not a supported operation", clientConfigurations.getOperation()));
+        }
+    }
+
+    private void invokeFunction(Exchange exchange) {
+
+        // convert exchange body to Map object
+        Object body = exchange.getMessage().getBody();
+        Map request;
+        if (body == null) {
+            if (LOG.isErrorEnabled()) {
+                LOG.error("Exchange body is null");
+            }
+            throw new IllegalArgumentException("Exchange body is mandatory and should be a valid Map or JSON string");
+        } else if (body instanceof Map) {
+            request = exchange.getMessage().getBody(Map.class);
+        } else {
+            String strBody = exchange.getMessage().getBody(String.class);
```

Comment:
Here you may face the case where the body is not a `String` nor `Map`

Reply:
Handled